x/oracle/abci_test.go

```diff
@@ -0,0 +1,170 @@
+package oracle_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/cosmos/cosmos-sdk/x/staking/teststaking"
+	"github.com/stretchr/testify/suite"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+	tmrand "github.com/tendermint/tendermint/libs/rand"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	ojoapp "github.com/ojo-network/ojo/app"
+	appparams "github.com/ojo-network/ojo/app/params"
+	"github.com/ojo-network/ojo/x/oracle"
+	"github.com/ojo-network/ojo/x/oracle/types"
+)
+
+const (
+	displayDenom string = appparams.DisplayDenom
+	bondDenom    string = appparams.BondDenom
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+
+	ctx sdk.Context
+	app *ojoapp.App
+}
+
+const (
+	initialPower = int64(10000000000)
+)
+
+func (s *IntegrationTestSuite) SetupTest() {
```

Comment:
Pls add a comment on why this SetupTest is unique - also should it be public?

Reply:
Its defined by the suite package so it needs the same function signature. If its changed to private it wont run before the tests.